gapic/templates/tests/unit/gapic/%name_%version/%sub/test_macros.j2

```diff
@@ -1947,3 +1929,91 @@ def test_unsupported_parameter_rest_asyncio():
     )
 
 {% endmacro %}
+
+{# is_rest_unsupported_method renders:
+ # 'True' if transport is async REST.
+ # 'True' if transport is sync REST and method is a client streaming method.
+ # 'False' otherwise.
+#}
+{# NOTE: We will keep updating this method as we add support for methods in async REST. #}
+{% macro is_rest_unsupported_method(method, is_async) %}
+{%- if method.client_streaming or is_async -%}
+{{'True'}}
+{%- else -%}
+{{'False'}}
+{%- endif -%}
+{% endmacro %}
+
+{# rest_transport_specific_tests generates all the rest specific tests for both
+# sync and async transport. It refactors the test cases in macro::rest_required_tests
+# and should replace it once we cover all the test cases in it here.
+#}
+{% macro rest_transport_specific_tests(service, transport, is_async) %}
+{% if 'rest' in transport %}
+{### Test syntax configuration begins ###}
+{% set await_prefix = "await " if is_async else "" %}
+{% set async_prefix = "async " if is_async else "" %}
+{% set async_decorator = "@pytest.mark.asyncio " if is_async else "" %}
```

Comment:
I wonder if these should be converted to macros so we can use them globally, similar to how you have `is_rest_unsupported_method`. We could remove the whole `{### Test syntax configuration begins ###}` block. WDYT?

Reply:
addressed.